### PR TITLE
Adding in the UI tests

### PR DESCRIPTION
**Pull request by rmbielby**
Sorry, it's been a bit slow getting this in! Here's the basic UI testing set up in shinytest. It still needs a few little bits of tidying up, but I'm just wanted to open it up for review at this stage to give you both sight. The tests go through each tab individually and check the output based on a handful of different input combinations for each one. The regional tab creates a bulk load of map data in the output, so that's filtered out of any checks. 

I started off looking into unit tests, but didn't seem to be so many 'unit' level components there to test, so shifted on to UI tests. Happy to look into getting unit tests added in as well.

### CODE REVIEW

File: global.R [hunk 1]
```diff
@@ -1,31 +1,54 @@
...
+shhh(library(shinyWidgets))
+library(shinytest)
+shhh(library(networkD3))
+shhh(library(reactable))
+shhh(library(knitr))
+shhh(library(Hmisc))
+shhh(library(matrixStats))
+shhh(library(sourcetools))
+shhh(library(shinyBS))
 library(dplyr)
```
Comment (cjrace) [inline global.R]: Missing shhh() on a couple of packages

File: tests/shinytest/testUI.R [hunk 2]
```diff
@@ -0,0 +1,225 @@
+app <- ShinyDriver$new("../../", loadTimeout = 1e+05)
+
+app$snapshotInit("testUI", screenshot = TRUE)
```
Comment (cjrace) [inline tests/shinytest/testUI.R]: I'd change this to screenshot = FALSE as the screenshots create false positives on tests on other machines (my computer had differences in the png's but not the jsons)

**Review by cjrace — CHANGES_REQUESTED** [review]
Couple of small changes to make, and then we need to think about how we're merging this with Charlotte's PR as it's included tidied code so will have created plenty of conflicts

**Comment by cjrace** [comment]
Noticed a concerning thing in the logs that shinytest seems to have failed but the action has passed - looks like we need to look at the code for the GitHub actions to make sure we capture when there's differences found and throw an error out. However we fix this, we should retest all other repos using this (off the top of my head it's just the template, SCAP, and Screener)

File: tests/shinytest/testUI.R (hunk 2 above)
Comment (rmbielby) [reply tests/shinytest/testUI.R]: Done

File: global.R (hunk 1 above)
Comment (rmbielby) [reply global.R]: Done

**Comment by cjrace** [comment]
Includes #1 #3 #7

**Comment by cjrace** [comment]
Closing in favour of #25 and #27

**Pull request closed without merging**